src/models/SpecificationFile.ts

```diff
@@ -12,14 +13,52 @@ const allowedFileNames: string[] = [
 ];
 const TYPE_CONTEXT_NAME = 'context-name';
 const TYPE_FILE_PATH = 'file-path';
+const TYPE_URL_PATH = 'url-path';
+
+export class Specification {
+  private readonly spec: string;
+  private readonly filePath?: string;
+  private readonly URLPath?: string;
+  constructor(spec: string, options?: { filepath?: string, URLPath?: string }) {
+    this.spec = spec;
+    this.filePath = options?.filepath;
+    this.URLPath = options?.URLPath;
+  }
+
+  text() {
+    return this.spec;
+  }
+
+  getFilePath() {
+    return this.filePath;
+  }
+
+  getURLPath() {
```

Comment:
I think you shouldn't call this "URL path". That's a URL.

I mean, this is a URL: `https://www.asynapi.com/blog`.
And this is the URL path: `/blog`.

They're not the same and I think here you mean the full URL, not just the path part.

Why not `getFileURL()`?

```suggestion
  getFileURL() {
```

If you change it, you'd have to change everything else following similar notation, like `const TYPE_URL_PATH = 'url-path';`